code/modules/mob/living/basic/farm_animals/mushroom.dm

```diff
@@ -0,0 +1,185 @@
+/mob/living/basic/mushroom
+	name = "walking mushroom"
+	desc = "It's a massive mushroom... with legs?"
+	icon_state = "mushroom_color"
+	icon_living = "mushroom_color"
+	icon_dead = "mushroom_dead"
+	mob_biotypes = MOB_ORGANIC | MOB_PLANT
+	butcher_results = list(/obj/item/food/hugemushroomslice = 1)
+
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "whacks"
+	response_harm_simple = "whack"
+
+	speed = 1
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomp"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+
+	faction = list(FACTION_MUSHROOM)
+	speak_emote = list("squeakes")
+	death_message = "fainted!"
+
+	ai_controller = /datum/ai_controller/basic_controller/mushroom
+	var/cap_color = "#ffffff"
+	///Tracks our general strength level gained from eating other shrooms
+	var/powerlevel = 0
+	///If someone tries to cheat the system by attacking a shroom to lower its health, punish them so that it won't award levels to shrooms that eat it
+	var/bruised = FALSE
+	///If we hit three, another mushroom's gonna eat us
+	var/faint_ticker = 0
+	///Where we store our cap icons so we dont generate them constantly to update our icon
+	var/static/mutable_appearance/cap_living
+	///Where we store our cap icons so we dont generate them constantly to update our icon
+	var/static/mutable_appearance/cap_dead
+	///So you can't repeatedly revive it during a fight
+	COOLDOWN_DECLARE(recovery_cooldown)
+
+/mob/living/basic/mushroom/Initialize(mapload)
+	melee_damage_lower = rand(4, 6)
+	melee_damage_upper = rand(11,21)
+	maxHealth = rand(50,70)
+	cap_living = cap_living || mutable_appearance(icon, "mushroom_cap")
+	cap_dead = cap_dead || mutable_appearance(icon, "mushroom_cap_dead")
+	cap_color = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+	UpdateMushroomCap()
+	health = maxHealth
+	. = ..()
```

Comment:
Why are we calling parent in the middle? Was the code always like this? Try moving it to the top of the proc and see if anything breaks and we can go from there, but this seems to be completely un-necessary for the basic mob framework and what we do before the parent call.